hostapd: add ht and vht support in handle event function

PolynomialDivision opened the pull request:
Add ht and vht support. If a device sends a probe request, the ht and vht fields indicate if the device can use ht and vht.

Signed-off-by: Nick Hainke vincent@systemli.org

nbd168 commented:
Please squash all commits into one (especially since commit 1 is reverted and applied differently in commit 3, and commit 2 is useless)

PolynomialDivision committed:
hostapd: add ht and vht support in handle event function
Add ht and vht capabilities. If a device sends a probe request,
the capabilities are added.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
@nbd168 Done. :)

PolynomialDivision commented:
@nbd168 Can you merge this pull request? :)

jow- commented:
Wouldn't it be better to expose the capability values as-is? If one is interested in the actual caps, he would need to extend or change the interface yet again.

PolynomialDivision commented:
@jow- This creates more overhead but that are intersting information...
Maybe we could do it like the [notify_response flag](https://github.com/PolynomialDivision/openwrt/blob/1419aa4dd85479474affa7bc910f58d361d71933/package/network/services/hostapd/src/src/ap/ubus.h#L41) in the hostapd_ubus_bss.
So a subscriber can choose which information he actually wants...
Furthermore, I don't get it why the response flag is an `int` and not `u8`. I think this should be changed to save memory?

blogic commented:
merged, Thanks !
